haskell/src/Ligo/BaseDAO/Types.hs

```diff
@@ -509,19 +509,13 @@ type CustomEntrypoint = (MText, ByteString)
 type FreezeParam = ("amount" :! Natural)
 type UnfreezeParam = ("amount" :! Natural)
 
--- NOTE: Constructors of the parameter types should remain sorted for the
---'ligoLayout' custom derivation to work.
---
--- TODO #259: The comment above should no longer be the case since the following
--- issue has been fixed:
--- https://gitlab.com/morley-framework/morley/-/issues/527
 data ForbidXTZParam
   = Drop_proposal ProposalKey
-  | Flush Natural
-  | Freeze FreezeParam
-  | Unfreeze UnfreezeParam
-  | Update_delegate [DelegateParam]
   | Vote [PermitProtected VoteParam]
+  | Update_delegate [DelegateParam]
+  | Unfreeze UnfreezeParam
+  | Freeze FreezeParam
+  | Flush Natural
```

Comment:
Instead of using another random sorting I think it would be more useful to have this match the order in which the contructors of the relative Ligo type are defined in `types.mligo`:
```suggestion
  | Vote [PermitProtected VoteParam]
  | Flush Natural
  | Freeze FreezeParam
  | Unfreeze UnfreezeParam
  | Update_delegate [DelegateParam]
```

Reply:
Related: it's possible that we did something similar in other types in this module, can you check the other places in which `ligoLayout` was used?

Reply:
I will do it. But I didn't think it mattered because the sorting of constructors does not matter at all now, so it was fine to leave the rest at what ever current ordering they are in. And by shuffling one type, we got proof that it works as expected.

Reply:
Yeah, that's a good point, reordering the other types won't prove much more.

I was thinking however that it wouldn't hurt either and generally it's easier to deal with changes on the two "sides" when things are lined up.

Reply:
Done.